feat(byol): Add BringYourOwnLinux documentation (#5664)
ovh/docs: pages/bare_metal_cloud/dedicated_servers/bring-your-own-linux/guide.en-gb.md

- Access to the [OVHcloud Control Panel](https://ca.ovh.com/auth/?action=gotomanager&from=https://www.ovh.com/world/&ovhSubsidiary=we) (for the method "[Deployment via Control Panel](#viacontrolpanel)")
- Access to the [OVHcloud API](/pages/manage_and_operate/api/first-steps) (for the section "[Deployment via API](#viaapi)" of this guide)
- Your image must be smaller than the Server RAM minus 3GiB

> [!warning]
>
> A new installation with BYOLinux will erase all the data on the server.
>

## Instructions

**Technical limitations**

There are some technical limitations linked to the use of physical products such as dedicated servers. Here is a non-exhaustive list, to keep in mind during your deployment preparation:

- Boot type: **uefi** or **legacy**
- Partition type: **MBR** or **GPT**
- Image format: **qcow2**

**Deployment methods**

- [Deployment via the Control Panel](#viacontrolpanel): allows you to simply deploy your image using the OVHcloud Control Panel.
- [Deployment via API](#viaapi): you can use the OVHcloud API to integrate images into your own scripts to automate deployments.

### Deploy your image via the Control Panel <a name="viapanel"></a>

> ***TBD***

Reviewer: ![image](https://github.com/ovh/docs/assets/17429937/2cfa8207-5d03-43b0-999c-0f19662ca246)
![image](https://github.com/ovh/docs/assets/17429937/9e89cf79-1539-478e-a7cd-8aba4c11b30c)

Reviewer: You can use and adapt this text: https://github.com/ovh/docs/blob/fix/PUBM-35205-byoi/pages/bare_metal_cloud/dedicated_servers/bring-your-own-image/guide.en-gb.md?plain=1#L57-L75

Reviewer: ![2024-01-25_16-50](https://github.com/ovh/docs/assets/17429937/431c24ba-8fb2-4abb-bf87-7609afa16650)
![2024-01-25_16-53](https://github.com/ovh/docs/assets/17429937/45b40823-65fc-4017-b3da-f05c240c4aba)

Reviewer: Without adding the screenshot for partitioning customization, maybe say that it's possible + link to https://help.ovhcloud.com/csm/en-dedicated-servers-getting-started-dedicated-server?id=kb_article_view&sysparm_article=KB0043477 ?

![BringYourOwnLinux Control Panel 02](images/byolinux-controlpanel02.png){.thumbnail}

In the windoe that appears, select `CUSTOM`, in the menu, then `Bring Your Own Linux - byolinux`, and click on `Next`{.action}.

![BringYourOwnLinux Control Panel 03](images/byolinux-controlpanel03.png){.thumbnail}

You will be redirected to the configuration page. Make sure your image URL is in the correct format. Complete the rest of the required fields on this page. Once you have confirmed that the information is correct, click `Confirm`{.action}.

You can find more details on the options in the "[Deployment options](#options)" section below.

For more information and examples about Cloud-Init's ConfigDrive, please consult the official documentation on [this page](https://cloudinit.readthedocs.io/en/22.1_a/topics/examples.html).

![BringYourOwnLinux Control Panel 04](images/byolinux-controlpanel04.png){.thumbnail}

### Deploy your image via the APIs <a name="viaapi"></a>

> [!api]
>
> @api {v1} /dedicated/server/{serviceName}/install/start POST /dedicated/server/{serviceName}/install/start
>

The base payload for the `install/start` is 

> ```json

Reviewer: remove this please, it will change very soon several times, I don't want/need to update this

Reviewer: (I'm going to talk about it in my os installation via API page)

    },
    {
      "key": "httpHeaders0Key",
      "value": "Authorization"
    },
    {
      "key": "httpHeaders0Value",
      "value": "Basic bG9naW46cGFzc3dvcmQ="
    },
    {
      "key": "imageCheckSum",
      "value": "367f26c915f39314dde155db3a2b0326803e06975d1f4be04256f8b591e38fd4062d36eb7d50e99da7a50b7f4cd69640e56a4ab93e8e0274e4e478e0f84b5d29"
    },
    {
      "key": "imageCheckSumType",
      "value": "sha512"
    },
    {
      "key": "configDriveUserData",
      "value": "#cloud-config\nssh_authorized_keys:\n  - ssh-rsa AAAAB8djYiw== myself@mydomain.net\n\nusers:\n  - name: patient0\n    sudo: ALL=(ALL) NOPASSWD:ALL\n    groups: users, sudo\n    shell: /bin/bash\n    lock_passwd: false\n    ssh_authorized_keys:\n      - ssh-rsa AAAAB8djYiw== myself@mydomain.net\ndisable_root: false\npackages:\n  - vim\n  - tree\nfinal_message: The system is finally up, after $UPTIME seconds\n"
    }
  ]
}
```

#### Deployment options <a name="options"></a>

| Field | Description | Required |
|-|-|-|
| userMetadata/imageURL | your linux image URL | ✅ |
| userMetadata/imageCheckSum | Your image's checksum. | ❌ |

> | userMetadata/imageCheckSumType | Your image's checksum type (md5, sha1, sha256, sha512). | ❌ (only if checksum provided)|

Reviewer: ```suggestion
| userMetadata/imageCheckSumType | Your image's checksum type (md5, sha1, sha256, sha512). | ❌ (except if checksum provided)|
```

> | userMetadata/configDriveUserData | your configDrive file content | ❌¹|

Reviewer: ```suggestion
| userMetadata/configDriveUserData | your configDrive file content¹ | ❌ |
```

> | userMetadata/httpHeaders?Key | HTTP Headers key  | ❌² |

Reviewer: ```suggestion
| userMetadata/configDriveMetadata | Custom CloudInit metadatas  | ❌ |
| userMetadata/httpHeaders?Key | HTTP Headers key  | ❌² |
```

| userMetadata/httpHeaders?Value | HTTP Headers value | ❌² |

¹ Can either be a `#cloud-config` or a script. It must be in one-line, and have `\n` for line-return<br />
² Use only if you need HTTP Headers, such as Basic Auth<br />

The ConfigDrive partition is used by cloud-init during the first server boot in order to apply your configurations. You can choose whether you want to use the default one, or a custom one (using `configDriveUserData`).

After you have completed the fields, start the deployment by clicking `Execute`{.action}.

## Go further

> Check the extensive explanation at [BringYourOwnLinux](https://github.com/ovh/BringYourOwnLinux)<br />

Reviewer: ```suggestion

[Check the extensive explanation](https://github.com/ovh/BringYourOwnLinux)

[OVHcloud API & Partitioning](/pages/bare_metal_cloud/dedicated_servers/partitioning_ovh)

[OVHcloud API & OS installation](/pages/bare_metal_cloud/dedicated_servers/api-os-installation)

[Bring Your Own Image (BYOI)](/pages/bare_metal_cloud/dedicated_servers/bring-your-own-image)

[Bring Your Own Image (BYOI) / Bring Your Own Linux (BYOLinux), a comparison sheet](pages/bare_metal_cloud/dedicated_servers/bring-your-own-image-versus-bring-your-own-linux)

```
